8월5일 금요일 모의고사 1문제 품(1문제 될듯말듯)

DrugCoding opened the pull request.

hphktutor3 commented on 1회차/이현성/0_민석이의과제체크하기.py:
```diff
@@ -1,3 +1,17 @@
+            student.pop(good_stu)
+
+
+    print(f'{i+1} {student}, end = ""')
```
음....현성님의 '민석이의 과제 체크하기'는 맞았다고 체크가 되어있는데, 코드를 보니 무언가 버전관리가 잘 되지 않은 것 같습니다. 이상해서 코드를 동일하게 제출해봐도 틀렸다고 나오네요 ㅠㅠ 안타깝게도 원래 푸신 코드의 리뷰를 해드리기는 어렵지만 여기 있는 코드에서 리뷰를 해 드리겠습니다.

우선 과제를 제출한 학생을 입력받고, 1부터 b까지 반복하면서 good_stu에 포함되는지 안되는지를 바탕으로 판단하려고 하셨던 것 같습니다. 뒷부분의 코드가 조금 달라질 수 있겠지만 저라면 반복문 내에서
if i not in good_stu: print(i, end=' ')
처럼 코드를 작성해서
바로바로 정답을 출력할 것 같습니다. 코드가 최종 코드가 아닌 것 같아 제대로 된 리뷰를 드리지 못해 죄송합니다 ㅠㅠ

hphktutor3 commented on 1회차/이현성/1_조교의성적매기기.py:
```diff
@@ -1,3 +1,34 @@
+        if cnt == c:
+            sum_ += 1
+                
+    print(f'#{t} {sum_}')
```
이 문제는 조교의 성적 매기기 문제인지 다시 한번 확인해볼 필요가 있을 것 같습니다. 제 생각에는 '어디에 단어가 들어갈 수 있을까'에 대한 문제 풀이인 것 같네요. '어디에 단어가 들어갈 수 있을까' 라고 생각하고 리뷰 드리겠습니다!! 코드가 진짜 너무 아깝게 틀렸네요 ㅠㅠ

어디에 단어가 들어갈 수 있을까의 핵심은 단어의 길이가 3인 경우에는 1이 정확하게 3번만 연속해야 하고 그 좌우로는 0 또는 범위를 벗어난 경우일 때에 일어나게 됩니다. 따라서 matrix가 1이기 때문에 cnt를 1 추가하고 그 뒤로 cnt==c 이더라도 그 다음에 1이 또 나올 수 있기 때문에 바로 sum을 더해주시면 틀리게 되는 것입니다. 그 조건문을 제거하고 차라리 1이 아닌 else 부분에서 cnt==c라면 sum을 추가하고 cnt=0으로 바꿔주는 코드로 만들었다면 정답이 나오게 될 겁니다. 한번 코드 바꿔보시고 제출해보시면 좋겠네요.
너무 잘하셨는데, 아주 아까웠던 풀이였습니다. 잘하셨습니다!

hphktutor3 replied:
다른 문제들을 다시 복습하실 때 혹시 접근법을 잘 모르실까봐 힌트를 조금씩 드리도록 하겠습니다 


- 조교의 성적 매기기
조교의 성적 매기기는 2개의 작은 문제로 쪼갤 수 있습니다. 모든 학생에 대해 3개의 숫자를 입력받아 비율에 맞춰 총점을 구하는 문제입니다. For 문, 또는 list comprehension 등 원하시는 방법을 통해 구하시면 됩니다.
다음으로는 K번째 학생의 등수를 확인하고, 학점을 출력해야 합니다. 등수만 알면 학점은 비교적 쉽게 구할 수 있을 것입니다. 등수는 sorted함수를 적절히 사용하면 K번째 학생의 점수가 몇 번째 인덱스에 위치하고 있는지 확인할 수 있을 것입니다. 등수를 확인할 때에는 sorted().index()를 쓰면 편하다는 점을 기억하시면 좋을 것 같네요 ㅎㅎ


- 파리퇴치
이차원리스트 부분은 KDT의 짧은 수업 시간동안에 익숙해지기 정말 어려운 일입니다. 이차원 리스트를 많이 접해보고 다양한 문제에서 이차원 리스트를 순회해보는 등 많은 연습이 필요합니다. 막막할 수 있겠지만, 단순히 노력만 하면 누구나 경지에 오를 수 있다고 생각하면 충분히 가치있는 공부가 될 것입니다.
파리퇴치는 모든 M*M 크기의 직사각형의 합을 구할 수만 있다면 나머지는 어렵지 않을 것입니다. 어떻게 반복문을 구성해야 하는지가 어려운 부분인데, 우선 합을 구하고자 하는 사각형의 제일 좌상단 위치가 (x 번째 행, y번째 열)라고 할 때, 합을 구해보고, (x,y)를 바꿔가면서 합을 구하는 방식이 가장 정석적인 풀이입니다.

- 암호생성기
암호 생성기 문제는 문제 자체의 이해가 어려우셨을 것 같네요. 정리해보면, 숫자 8개를 받아서 맨 앞의 숫자에다 1~5를 빼고 맨 뒤로 옮기는 작업을 무한히 반복하다가 어느 순간 0 또는 음수가 되는 시점에 그 작업을 그만두는 문제이구나 하고 이해하시면 됩니다. 문제 이해가 어려웠지만 앞에서 빼서 뒤에 넣는 작업을 보니 Queue(Deque)를 사용하는 문제라는 느낌이 오셨을 겁니다.
이후에는 while/break 문을 적절히 활용하여 무한히 문제의 작업을 계속 반복해주시면 됩니다.
또한 이 문제는 test_case의 총 개수를 알 수 없는데, 이럴 때에는 try: input() / except: break 하여 입력을 받으시면 해결될 것입니다.

- 괄호 짝짓기
괄호 짝짓기 문제는 아마 이전에 풀어본 경험이 있거나 코드리뷰로 접하셨을 겁니다. 다만 괄호의 종류가 많다는 점에서 복잡해보일 뿐이지요. 풀이법은 똑같습니다. Stack 자료구조를 활용해 여는 괄호면 종류에 관계없이 stack에 추가하고, 닫는 괄호면 stack이 비어있는지, pop의 결과와 짝이 맞는지 등을 확인하면서 유효 여부를 파악하시면 됩니다.
코드의 구현이 조금 어렵다면 이전에 푸셨던 문제를 다시 한 번 풀어보시면서 공부하셔도 좋을 것 같네요 :)

hphktutor3 commented on 1회차/이현성/README.md:
```diff
@@ -1 +1,17 @@
+
+푸는 방법까지 아예 다 모르면 마음이라도 편한데
+
+코드 구현을 마음대로 못하니 더 화가 나고 답답합니다
```
다른 튜터님들과 마찬가지로 저 또한 낮은 단계의 문제를 반복하면서 풀어보는 것이 굉장히 큰 도움이 될 것이라고 생각합니다. 백준 브론즈에도 생각보다 어려운 문제들이 많으니 예전에 풀었던 Codeup에서 다시 연습해보시는 것도 방법이겠지요..

아무래도 많은 사람들이 함께하는 수업이다보니 누구는 잘하고 누구는 못하고 비교가 되기 쉽습니다. 조원 분들에 비해서 나의 코드 구현이 깔끔하지 못한 것 같고 비교되고 실력이 늘지 않는듯한 기분. 저 또한 예전에 많이 느꼈던 감정들입니다. 저는 매너리즘에 빠져 알고리즘을 2~3년정도 그만두기도 했습니다. ㅎㅎ 그러다 갑자기 다시 해보니 또 손에 잡히기도 하더라고요. 좀 더 일찍 알고리즘을 했다면?? 하는 후회도 들지만 매너리즘에 빠지지 않았다면 이렇게 알고리즘을 열심히 공부하지도 않았을 겁니다.

때로는 비교를 멈추고 잠시 쉬는 것도 방법입니다. 결국에는 다양한 문제들을 풀어보시는 것이 실력향상에 도움이 되겠지만, 과연 알고리즘이 컴퓨터의 전부일까요? 현성님도 몇 주만 지나면 정말 다른 세계의 프로그래밍 공부를 하게 되실 겁니다. 알고리즘과는 또 다른 문제들을 접할테지요. 저는 알고리즘은 정말 잘하지만 이후 배울 프론트엔드 벡엔드(지금은 모르셔도 됩니다.) 분야는 정말 까막눈입니다. 지금도 꾸준히 공부중이긴 하지만 여전히 제 마음에는 들지 않습니다.

알고리즘을 지금 좌절하기에는 너무 이르다고 생각합니다. 알고리즘에서 좌절한다고 KDT에서 좌절하는 일은 더더욱 바보같은 일이고요. 더 슬픈 이야기일 수 있지만 나중에 코딩테스트를 실전으로 공부하시다보면 지금 배우는 알고리즘은 정말 티끌에 불과하다는 것을 느끼실 겁니다. 저희는 알고리즘만큼은 3주 동안 모든 부분을 다뤄드리기 어려우니 스스로 공부하는 법을 알려드리고 있습니다. 말 그대로 알고리즘 세계에서 걸음마를 떼도록 도와드리는 거죠. 걸음마를 떼기 전부터 좌절하시는 수강생 분들을 너무 많이 봐왔고, 그들의 후회도 심심치 않게 들려옵니다. 세상에서 제일 빠른 우사인볼트가 세상에서 제일 걸음마를 먼저 뗐을까요? 저는 현성님도 인내의 시간을 지나면 분명히 결실을 맺을 거라고 믿고 있습니다.

제 생각과 진심을 녹여 격려를 드렸지만 마음에 와 닿으셨을지 모르겠네요. 제가 너무 스트레스 받으시는 길로 인도하는 것이 아닐까 걱정되기도 합니다. 다른 것 보다도 현성님께서 편한 마음으로 어제보다 하나라도 더 배워가며 재미를 느끼시길 기도하겠습니다. 화이팅입니다!

DrugCoding commented:
조언과 리뷰 너무 감사드립니다!!!
조교문제는 스스로도 너무 아쉬웠는데 아쉬웠다고 해주시니 뭔가 기운이 조금 났습니다!
잘 못따라가는 사람들을 위한 문제셋도 아마 준비해주시는 것 같은데
정말 비전공자를 위한 기초단단 문제모음도 있다면 받아보고 싶습니다..!
생각없이 계속 풀면서, 잘 풀리면서 기본적인 코드의 형태, 들여쓰기, 문법 등이 익숙해지고 싶습니다!

2022년 8월 9일 (화) 오후 10:04, hphktutor3 ***@***.***>님이 작성:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In 1회차/이현성/0_민석이의과제체크하기.py
> <https://github.com/kdt-hphk/01-PJT-04/pull/67#discussion_r941311349>:
>
> > @@ -1,3 +1,17 @@
>  import sys
>
>  sys.stdin = open("_민석이의과제체크하기.txt")
> +
> +a = int(input())
> +
> +for i in range(a):
> +    student = []
> +    b, c = map(int, input().split())
> +    good_stu = [map(int, input().split()) for i in range(b)]
> +    for i in range(1, b + 1):
> +        student.append(i)
> +        if good_stu in student:
> +            student.pop(good_stu)
> +
> +
> +    print(f'{i+1} {student}, end = ""')
>
> 음....현성님의 '민석이의 과제 체크하기'는 맞았다고 체크가 되어있는데, 코드를 보니 무언가 버전관리가 잘 되지 않은 것 같습니다.
> 이상해서 코드를 동일하게 제출해봐도 틀렸다고 나오네요 ㅠㅠ 안타깝게도 원래 푸신 코드의 리뷰를 해드리기는 어렵지만 여기 있는 코드에서
> 리뷰를 해 드리겠습니다.
>
> 우선 과제를 제출한 학생을 입력받고, 1부터 b까지 반복하면서 good_stu에 포함되는지 안되는지를 바탕으로 판단하려고 하셨던 것
> 같습니다. 뒷부분의 코드가 조금 달라질 수 있겠지만 저라면 반복문 내에서
> if i not in good_stu: print(i, end=' ')
> 처럼 코드를 작성해서
> 바로바로 정답을 출력할 것 같습니다. 코드가 최종 코드가 아닌 것 같아 제대로 된 리뷰를 드리지 못해 죄송합니다 ㅠㅠ
> ------------------------------
>
> In 1회차/이현성/1_조교의성적매기기.py
> <https://github.com/kdt-hphk/01-PJT-04/pull/67#discussion_r941311816>:
>
> > +            sum_ += 1
> +
> +
> +    for i in range(b):
> +        cnt = 0
> +        for j in range(b):
> +            if matrix[j][i] == 1:
> +                cnt += 1
> +                if cnt == c:
> +                    sum_ += 1
> +            else:
> +                cnt = 0
> +        if cnt == c:
> +            sum_ += 1
> +
> +    print(f'#{t} {sum_}')
>
> 이 문제는 조교의 성적 매기기 문제인지 다시 한번 확인해볼 필요가 있을 것 같습니다. 제 생각에는 '어디에 단어가 들어갈 수
> 있을까'에 대한 문제 풀이인 것 같네요. '어디에 단어가 들어갈 수 있을까' 라고 생각하고 리뷰 드리겠습니다!! 코드가 진짜 너무
> 아깝게 틀렸네요 ㅠㅠ
>
> 어디에 단어가 들어갈 수 있을까의 핵심은 단어의 길이가 3인 경우에는 1이 정확하게 3번만 연속해야 하고 그 좌우로는 0 또는 범위를
> 벗어난 경우일 때에 일어나게 됩니다. 따라서 matrix가 1이기 때문에 cnt를 1 추가하고 그 뒤로 cnt==c 이더라도 그 다음에
> 1이 또 나올 수 있기 때문에 바로 sum을 더해주시면 틀리게 되는 것입니다. 그 조건문을 제거하고 차라리 1이 아닌 else 부분에서
> cnt==c라면 sum을 추가하고 cnt=0으로 바꿔주는 코드로 만들었다면 정답이 나오게 될 겁니다. 한번 코드 바꿔보시고 제출해보시면
> 좋겠네요.
> 너무 잘하셨는데, 아주 아까웠던 풀이였습니다. 잘하셨습니다!
> ------------------------------
>
> In 1회차/이현성/README.md
> <https://github.com/kdt-hphk/01-PJT-04/pull/67#discussion_r941312988>:
>
> > +
> +이렇게 이렇게 해서 이렇게 계속 돌면서 저렇게 하면 되겠구나 생각은 잘하고
> +
> +구글링 해서 봐도 푸는 방법은 맞지만
> +
> +코드가 아직도 마음대로 구현이 되지 않습니다,,,(답답...)
> +
> +이 부분을 몇 번 반복 한다 여기서 몇 번을 반복한다 알고는 있지만
> +
> +코드를 쓰다 보면 헷갈리고 제대로 하지 못하는 것 같습니다
> +
> +같은 고민을 하고 있는 조원분이 튜터님이 그럴 때는 낮은 단계 문제를 반복해야 한다고 하는데 그게 맞을까요?
> +
> +푸는 방법까지 아예 다 모르면 마음이라도 편한데
> +
> +코드 구현을 마음대로 못하니 더 화가 나고 답답합니다
>
> 다른 튜터님들과 마찬가지로 저 또한 낮은 단계의 문제를 반복하면서 풀어보는 것이 굉장히 큰 도움이 될 것이라고 생각합니다. 백준
> 브론즈에도 생각보다 어려운 문제들이 많으니 예전에 풀었던 Codeup에서 다시 연습해보시는 것도 방법이겠지요..
>
> 아무래도 많은 사람들이 함께하는 수업이다보니 누구는 잘하고 누구는 못하고 비교가 되기 쉽습니다. 조원 분들에 비해서 나의 코드 구현이
> 깔끔하지 못한 것 같고 비교되고 실력이 늘지 않는듯한 기분. 저 또한 예전에 많이 느꼈던 감정들입니다. 저는 매너리즘에 빠져 알고리즘을
> 2~3년정도 그만두기도 했습니다. ㅎㅎ 그러다 갑자기 다시 해보니 또 손에 잡히기도 하더라고요. 좀 더 일찍 알고리즘을 했다면?? 하는
> 후회도 들지만 매너리즘에 빠지지 않았다면 이렇게 알고리즘을 열심히 공부하지도 않았을 겁니다.
>
> 때로는 비교를 멈추고 잠시 쉬는 것도 방법입니다. 결국에는 다양한 문제들을 풀어보시는 것이 실력향상에 도움이 되겠지만, 과연
> 알고리즘이 컴퓨터의 전부일까요? 현성님도 몇 주만 지나면 정말 다른 세계의 프로그래밍 공부를 하게 되실 겁니다. 알고리즘과는 또 다른
> 문제들을 접할테지요. 저는 알고리즘은 정말 잘하지만 이후 배울 프론트엔드 벡엔드(지금은 모르셔도 됩니다.) 분야는 정말 까막눈입니다.
> 지금도 꾸준히 공부중이긴 하지만 여전히 제 마음에는 들지 않습니다.
>
> 알고리즘을 지금 좌절하기에는 너무 이르다고 생각합니다. 알고리즘에서 좌절한다고 KDT에서 좌절하는 일은 더더욱 바보같은 일이고요. 더
> 슬픈 이야기일 수 있지만 나중에 코딩테스트를 실전으로 공부하시다보면 지금 배우는 알고리즘은 정말 티끌에 불과하다는 것을 느끼실 겁니다.
> 저희는 알고리즘만큼은 3주 동안 모든 부분을 다뤄드리기 어려우니 스스로 공부하는 법을 알려드리고 있습니다. 말 그대로 알고리즘 세계에서
> 걸음마를 떼도록 도와드리는 거죠. 걸음마를 떼기 전부터 좌절하시는 수강생 분들을 너무 많이 봐왔고, 그들의 후회도 심심치 않게
> 들려옵니다. 세상에서 제일 빠른 우사인볼트가 세상에서 제일 걸음마를 먼저 뗐을까요? 저는 현성님도 인내의 시간을 지나면 분명히 결실을
> 맺을 거라고 믿고 있습니다.
>
> 제 생각과 진심을 녹여 격려를 드렸지만 마음에 와 닿으셨을지 모르겠네요. 제가 너무 스트레스 받으시는 길로 인도하는 것이 아닐까
> 걱정되기도 합니다. 다른 것 보다도 현성님께서 편한 마음으로 어제보다 하나라도 더 배워가며 재미를 느끼시길 기도하겠습니다. 화이팅입니다!
> ------------------------------
>
> In 1회차/이현성/1_조교의성적매기기.py
> <https://github.com/kdt-hphk/01-PJT-04/pull/67#discussion_r941312728>:
>
> > +            sum_ += 1
> +
> +
> +    for i in range(b):
> +        cnt = 0
> +        for j in range(b):
> +            if matrix[j][i] == 1:
> +                cnt += 1
> +                if cnt == c:
> +                    sum_ += 1
> +            else:
> +                cnt = 0
> +        if cnt == c:
> +            sum_ += 1
> +
> +    print(f'#{t} {sum_}')
>
> 다른 문제들을 다시 복습하실 때 혹시 접근법을 잘 모르실까봐 힌트를 조금씩 드리도록 하겠습니다
>
>    -
>
>    조교의 성적 매기기
>    조교의 성적 매기기는 2개의 작은 문제로 쪼갤 수 있습니다. 모든 학생에 대해 3개의 숫자를 입력받아 비율에 맞춰 총점을
>    구하는 문제입니다. For 문, 또는 list comprehension 등 원하시는 방법을 통해 구하시면 됩니다.
>    다음으로는 K번째 학생의 등수를 확인하고, 학점을 출력해야 합니다. 등수만 알면 학점은 비교적 쉽게 구할 수 있을 것입니다.
>    등수는 sorted함수를 적절히 사용하면 K번째 학생의 점수가 몇 번째 인덱스에 위치하고 있는지 확인할 수 있을 것입니다. 등수를
>    확인할 때에는 sorted().index()를 쓰면 편하다는 점을 기억하시면 좋을 것 같네요 ㅎㅎ
>    -
>
>    파리퇴치
>    이차원리스트 부분은 KDT의 짧은 수업 시간동안에 익숙해지기 정말 어려운 일입니다. 이차원 리스트를 많이 접해보고 다양한
>    문제에서 이차원 리스트를 순회해보는 등 많은 연습이 필요합니다. 막막할 수 있겠지만, 단순히 노력만 하면 누구나 경지에 오를 수 있다고
>    생각하면 충분히 가치있는 공부가 될 것입니다.
>    파리퇴치는 모든 M*M 크기의 직사각형의 합을 구할 수만 있다면 나머지는 어렵지 않을 것입니다. 어떻게 반복문을 구성해야
>    하는지가 어려운 부분인데, 우선 합을 구하고자 하는 사각형의 제일 좌상단 위치가 (x 번째 행, y번째 열)라고 할 때, 합을
>    구해보고, (x,y)를 바꿔가면서 합을 구하는 방식이 가장 정석적인 풀이입니다.
>    -
>
>    암호생성기
>    암호 생성기 문제는 문제 자체의 이해가 어려우셨을 것 같네요. 정리해보면, 숫자 8개를 받아서 맨 앞의 숫자에다 1~5를 빼고
>    맨 뒤로 옮기는 작업을 무한히 반복하다가 어느 순간 0 또는 음수가 되는 시점에 그 작업을 그만두는 문제이구나 하고 이해하시면 됩니다.
>    문제 이해가 어려웠지만 앞에서 빼서 뒤에 넣는 작업을 보니 Queue(Deque)를 사용하는 문제라는 느낌이 오셨을 겁니다.
>    이후에는 while/break 문을 적절히 활용하여 무한히 문제의 작업을 계속 반복해주시면 됩니다.
>    또한 이 문제는 test_case의 총 개수를 알 수 없는데, 이럴 때에는 try: input() / except: break
>    하여 입력을 받으시면 해결될 것입니다.
>    -
>
>    괄호 짝짓기
>    괄호 짝짓기 문제는 아마 이전에 풀어본 경험이 있거나 코드리뷰로 접하셨을 겁니다. 다만 괄호의 종류가 많다는 점에서 복잡해보일
>    뿐이지요. 풀이법은 똑같습니다. Stack 자료구조를 활용해 여는 괄호면 종류에 관계없이 stack에 추가하고, 닫는 괄호면
>    stack이 비어있는지, pop의 결과와 짝이 맞는지 등을 확인하면서 유효 여부를 파악하시면 됩니다.
>    코드의 구현이 조금 어렵다면 이전에 푸셨던 문제를 다시 한 번 풀어보시면서 공부하셔도 좋을 것 같네요 :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kdt-hphk/01-PJT-04/pull/67#pullrequestreview-1066682500>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWASHLM53JYLJQZ4IO2SG2LVYJJPHANCNFSM55VHVTBQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
